outputs.tf

```diff
@@ -60,12 +60,19 @@ output "replication_group_cluster_mode_enabled" {
 }
 
 output "replication_group_endpoint_address" {
-  value       = var.cluster_mode_enabled ? aws_elasticache_replication_group.this[0].configuration_endpoint_address : aws_elasticache_replication_group.this[0].primary_endpoint_address
+  value       = length(aws_elasticache_replication_group.this) > 0 ? (var.cluster_mode_enabled ? aws_elasticache_replication_group.this[0].configuration_endpoint_address : aws_elasticache_replication_group.this[0].primary_endpoint_address) : ""
   description = "Address of the replication group endpoint."
 }
 
 output "replication_group_reader_endpoint_address" {
-  value       = var.cluster_mode_enabled ? "" : aws_elasticache_replication_group.this[0].reader_endpoint_address
+  value       = length(aws_elasticache_replication_group.this) > 0 ? (var.cluster_mode_enabled ? "" : aws_elasticache_replication_group.this[0].reader_endpoint_address) : ""
```

Comment:
* I see `resource` length checking for the first time. There is no need for this. Return as it was.